### PR TITLE
Offline acc

### PR DESCRIPTION
**Pull request by DenysGonchar**
ensuring that mod_offline preserves non-strippable fields of the accumulator

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7693.1](https://travis-ci.org/esl/MongooseIM/jobs/633889688) / Erlang 22.0 / small_tests / 6da03b0c0dfefb6d9ef8fd5b2f12ed2e18d57291
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2587/7693/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2587/7693/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-01-07_19.03.07/lib.mongooseim.logs/run.2020-01-07_19.03.08/suite.log.html)

**Comment by mongoose-im** [comment]
[7709.1](https://travis-ci.org/esl/MongooseIM/jobs/634403051) / Erlang 22.0 / small_tests / aecd3eda6f2afd58c82bd68c7be279f019fd681c
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2587/7709/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2587/7709/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-01-08_20.54.10/lib.mongooseim.logs/run.2020-01-08_20.54.11/suite.log.html)

**Comment by mongoose-im** [comment]
[7711.3](https://travis-ci.org/esl/MongooseIM/jobs/634652079) / Erlang 22.0 / riak_mnesia / 39012ebfcd2f83634ff841665110b2d91e22cb71
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2587/7711/riak_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2587/7711/riak_mnesia.22.0/big/ct_run.test%40travis-job-7a765acf-16fa-4e01-9eca-7910908908e4.2020-01-09_09.38.57/index.html)
**OK: 1469** / Failed: 0 / **User-skipped: 153** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[7712.2](https://travis-ci.org/esl/MongooseIM/jobs/634687015) / Erlang 22.0 / mysql_redis / 942a061c13999fbe737a3c37db9c5d5ee7bb7244
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2587/7712/mysql_redis.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2587/7712/mysql_redis.22.0/big/ct_run.test%40travis-job-9be957a8-9df1-4d49-8c31-d085ffe20a99.2020-01-09_11.07.50/index.html)
**OK: 162** / Failed: 0 / **User-skipped: 9** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[7713.2](https://travis-ci.org/esl/MongooseIM/jobs/634701723) / Erlang 22.0 / mysql_redis / 8ab37491ce1b48d49fa81b70832aeec00be36e72
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2587/7713/mysql_redis.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2587/7713/mysql_redis.22.0/big/ct_run.test%40travis-job-d5525e8f-880c-450e-a80a-618dc8bd863f.2020-01-09_11.47.09/index.html)
**OK: 162** / Failed: 0 / **User-skipped: 9** / Auto-skipped: 0

File: doc/migrations/3.5.0_3.5.0plus.md [hunk 1]
```diff
@@ -113,3 +113,28 @@ For more information, please check the specs for types and functions in the afor
 
 * The default room config is still the same, i.e. `roomname` (default: `"Untitled"`) and `subject` (empty string).
 * The room config representation in databases (both Mnesia and RDBMS) is the same; no need for migration.
+
+## Offline storage
+
+In this version offline storage entries contains one additional information for internal use.
```
Comment (goddammit) [inline doc/migrations/3.5.0_3.5.0plus.md]:
```suggestion
In this version the offline storage entries contain one additional information for internal use.
```

File: doc/migrations/3.5.0_3.5.0plus.md [hunk 2]
```diff
@@ -113,3 +113,28 @@ For more information, please check the specs for types and functions in the afor
...
+Riak and mnesia backends doesn't require any change when upgrading to this version.
```
Comment (goddammit) [inline doc/migrations/3.5.0_3.5.0plus.md]:
```suggestion
Riak and mnesia backends don't require any changes when upgrading to this version.
```

File: doc/migrations/3.5.0_3.5.0plus.md [hunk 3]
```diff
@@ -113,3 +113,28 @@ For more information, please check the specs for types and functions in the afor
...
+In case of RDBMS backends, new column needs to be added.
```
Comment (goddammit) [inline doc/migrations/3.5.0_3.5.0plus.md]:
```suggestion
In case of the RDBMS backends, a new column needs to be added.
```

**Comment by mongoose-im** [comment]
[7714.1](https://travis-ci.org/esl/MongooseIM/jobs/634722945) / Erlang 22.0 / small_tests / ea25b4bf091c9f6ddb4fe3e0c4b8bb451fdeb3e7
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2587/7714/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2587/7714/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-01-09_12.46.26/lib.mongooseim.logs/run.2020-01-09_12.46.26/suite.log.html)

**Pull request merged**